modulus/utils/capture.py

```diff
@@ -251,6 +253,13 @@ def _amp_forward(self, *args, **kwargs) -> Any:
         if not self.eval:
             # In training mode output should be the loss
             self.scaler.scale(output).backward()
+            if self.gradient_clip_norm is not None:
+                if self.use_autocast:
```

Comment:
I think this check is redundant - `GradScaler.unscale_` will check whether the scaler is enabled. Ok to keep it anyway for consistency with the other code.

Reply:
I will remove it, thanks.